app/src/marketplace/modules/dataUnion/services.js

```diff
@@ -160,66 +311,212 @@ export const getCommunityContract = (address: DataUnionId, usePublicNode: boolea
     }, usePublicNode)
 }
 
-export const getDataUnionOwner = async (address: DataUnionId, usePublicNode: boolean = false) => {
-    const contract = getCommunityContract(address, usePublicNode)
+export const getDataUnionVersion = async (address: DataUnionId, usePublicNode: boolean = true) => {
+    const client = createClient({
+        usePublicNode,
+    })
+    const version = await client.getDataUnionVersion(address)
+
+    return version || 0
+}
+
+// eslint-disable-next-line camelcase
+const deprecated_getDataUnionOwner = async (address: DataUnionId, usePublicNode: boolean = false) => {
+    const contract = deprecated_getCommunityContract(address, usePublicNode)
     const owner = await call(contract.methods.owner)
 
     return owner
 }
 
-export const isDataUnionDeployed = async (address: DataUnionId, usePublicNode: boolean = false) => (
-    !!getDataUnionOwner(address, usePublicNode)
-)
+export const getDataUnionOwner = async (address: DataUnionId, usePublicNode: boolean = false) => {
+    const version = await getDataUnionVersion(address, usePublicNode)
 
-export const getAdminFee = async (address: DataUnionId, usePublicNode: boolean = false) => {
+    if (version === 2) {
```

Comment:
Here I think it should be the client's job to handle the versions. Simply `return client.getAdminAddress({dataUnion: address})` should do the job, and if it doesn't, let's fix it! (true, currently it might not, so I'm not suggesting this be changed in this PR)

Reply:
(though for both the name of the getter function is `owner()` so it probably just Just Work... was it tested that it didn't?)